### PR TITLE
Add more platforms to CI

### PR DESCRIPTION
**Pull request by justsmth**
### Description of changes: 
* Add CI coverage for riscv64, armv6 and loongarch64.
* Update "Other Platforms" section of README.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,7 +109,9 @@ that improve the experience for consumers on these platforms.
 | Android   | arm32       |
 | iOS       | aarch64     |
 | Linux     | arm32       |
-| Linux     | Loongarch64 | 
+| Linux     | loongarch64 |
+| Linux     | risc-v64    |
```
Comment (andrewhop) [inline README.md:113]: Should this go up in supported platforms now that it's in the CI?

File: .github/workflows/cross-test.yml [hunk 2]
```diff
@@ -48,3 +48,43 @@ jobs:
...
+      - name: RISC-V 64 Build/Test
+# The flag below is set to avoid the following error with GCC 11.4.0:
+#
+#  /home/runner/work/aws-lc/aws-lc/crypto/pem/pem_lib.c:705:11: error: 'strncmp' of strings of length 1 and 9 and bound of 9 evaluates to nonzero [-Werror=string-compare]
+#  705 |       if (strncmp(buf, "-----END ", 9) == 0) {
+#      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        env:
+          CFLAGS: "-Wno-string-compare"
+        run: tests/ci/run_cross_tests.sh riscv riscv64-unknown-linux-gnu "-DCMAKE_BUILD_TYPE=Release"
+  armv6-non-fips-build-test:
```
Comment (andrewhop) [inline .github/workflows/cross-test.yml:68]: Also this raises the question: should we have an ARM version section to the readme?

File: README.md (hunk 1 above)
Comment (justsmth) [reply README.md:113]: I think the standard for a platform being listed in the previous section is a little higher than just being in the CI.  But this should be discussed further.

File: .github/workflows/cross-test.yml (hunk 2 above)
Comment (justsmth) [reply .github/workflows/cross-test.yml:68]: Perhaps it could be clarified.  The earliest `arm32` platform we can support might be `armv4`.

**Review by andrewhop — APPROVED** [review]

**Review by WillChilds-Klein — DISMISSED** [review]

**Review by WillChilds-Klein — APPROVED** [review]

**Review by skmcgrail — APPROVED** [review]

**Pull request merged**